auth/allow_all_authenticator.cc

```diff
@@ -33,6 +33,7 @@ static const class_registrator<
         authenticator,
         allow_all_authenticator,
         cql3::query_processor&,
-        ::service::migration_manager&> registration("org.apache.cassandra.auth.AllowAllAuthenticator");
+        ::service::migration_manager&,
+        authenticator_config> registration("org.apache.cassandra.auth.AllowAllAuthenticator");
```

Comment:
I've first tried the discuss approach of using a set_configuration method but at the end it was still requiring change on all authenticators leading to mostly same impact in term of number of changed to apply while I found this approach quite wrong in term of design as the configuration is mandatory. Using the set_config method requires then to ensure that it is called immediately after constructing the object.
If needed we can discuss this in IRL

Reply:
As discussed will give a try to both comment to avoid changing constructor.
I will push to specific commit for those changes so we can then assert if it well reduce scope of change